app/views/additional_details/_dashboard_complete_additional_details.html.haml

```diff
@@ -1,7 +1,9 @@
 - if service_request.additional_detail_services.present?
   - service_request.line_items.each do |line_item|
-    - unless line_item.submissions.present?
-      %div.alert.alert-warning
-        %h5.text-center
-          = link_to t("additional_details.dashboard_complete_details", service_name: Service.find(line_item.service_id).name),
-            new_service_additional_details_submission_path(line_item.service_id, protocol_id: @protocol.id, line_item_id: line_item.id), remote: true
+    - service = Service.find(line_item.service_id)
+    - if service.questionnaires.active.present?
+      - unless line_item.submissions.present?
```

Comment:
I think the expression in the `unless` should be tacked on to the `if` clause on line 4.